### PR TITLE
fix(app): stop click outside modal redirect 

### LINKED ISSUE #10277: 6.0 Feedback: clicking outside modal loads protocol detail page
### Overview

There are three ways to dismiss analysis error modals, one of which unexpectedly goes to the protocol detail page.

### Current Behavior

Clicking outside a protocol analysis error modal moves from the protocol list page to the detail page for that protocol. Clicking the "X" button at the top of the modal or "Close" at the bottom of the modal dismisses the modal properly, returning to the protocol list page (which is visible underneath the modal the entire time).

This also happens when clicking outside the "Delete Protocol" confirmation modal.

### Expected Behavior

All three methods of dismissing the modal should behave identically, returning to the immediately previous (and still visible) page.

As an alternative, clicking outside the modal could do nothing, i.e. require the user to click a button to dismiss it.

### Steps To Reproduce

For error modals:
1. Import a protocol that will fail analysis. 
[tabs_and_spaces.py.zip](https://github.com/Opentrons/opentrons/files/8682729/tabs_and_spaces.py.zip)
2. Click on "error details" in the banner that appears on the newly imported protocol.
3. Click anywhere outside the modal that appears.

For deletion modals:
1. Click the overflow menu on any protocol in the list, and choose "Delete protocol"
2. Click anywhere outside the modal that appears.

https://user-images.githubusercontent.com/1958332/168172548-8af7cc5b-3a93-4f7a-9b21-e3611b7dbefa.mov

### Operating system

Mac

### Robot setup or anything else?

_No response_

### PR DESCRIPTION
**Pull request by sakibh**
# Overview

This PR stops the redirect when clicking outside of the modal, particularly the failed protocol
analysis modal.

closes https://github.com/Opentrons/opentrons/issues/10277

# Changelog

- Add `stopPropogation` to the appropriate components/handlers

# Review requests

- Upload a protocol that fails analysis. Click the `error details` link in the banner which should open a modal. Then try clicking outside the modal and make sure that the app does NOT redirect to the protocol details page.

# Risk assessment

low

### CODE REVIEW

**Review by ecormany — COMMENTED** [review]
🎉 It works! I see we went with the option to do nothing when clicking outside the modal. Fine by me, we already have two close boxes.

Will let an engineer give final sign-off on the implementation.

File: app/src/organisms/ProtocolAnalysisFailure/index.tsx [hunk 1]
```diff
@@ -36,14 +36,18 @@ export function ProtocolAnalysisFailure(
 
   const handleClickShowDetails: React.MouseEventHandler = e => {
     e.preventDefault()
+    e.stopPropagation()
     setShowErrorDetails(true)
   }
   const handleClickHideDetails: React.MouseEventHandler = e => {
     e.preventDefault()
+    console.log('testing')
```
Comment (b-cooper) [inline app/src/organisms/ProtocolAnalysisFailure/index.tsx]:
```suggestion
```

File: app/src/atoms/Banner/index.tsx [hunk 2]
```diff
@@ -31,6 +31,7 @@ export interface BannerProps extends StyleProps {
   onCloseClick?: (() => unknown) | React.MouseEventHandler<HTMLButtonElement>
   /** Override the default Alert Icon */
   icon?: IconProps
+  onClick?: () => unknown
```
Comment (b-cooper) [inline app/src/atoms/Banner/index.tsx]: it doesn't look like we need this prop right?

**Review by jerader — APPROVED** [review]
lgtm!

**Pull request merged**